tests/Unit/Helpers/Domain/DomainTestHelpers.cpp

```diff
@@ -255,6 +257,85 @@ tnsr::I<double, VolumeDim, Frame::BlockLogical> point_in_neighbor_frame(
   return get_corner_of_orthant(point_get_orthant);
 }
 
+// This tests whether a logical grid point on the face of the host Block
+// corresponds to the same logical grid point on the abutting face of the
+// neighbor Block (taking into account the discrete rotation of the
+// OrientationMap from the host Block to the neighbor Block).
+template <size_t VolumeDim>
+void check_block_face_grid_points_align(
+    const Block<VolumeDim>& host_block, const Block<VolumeDim>& neighbor_block,
+    double time = std::numeric_limits<double>::signaling_NaN(),
+    const std::unordered_map<
+        std::string, std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime>>&
+        functions_of_time = {}) {
+  const auto direction = find_direction_to_neighbor(host_block, neighbor_block);
+  const auto orientation =
+      find_neighbor_orientation(host_block, neighbor_block);
+  // Set up a Mesh on the shared face. Corner points are already checked by
+  // physical_separation, so check on Gauss points
+  const Mesh<VolumeDim - 1> face_mesh{3_st, Spectral::Basis::Legendre,
+                                      Spectral::Quadrature::Gauss};
+  // logical coordinates on the face mesh in element logical coordinates
+  tnsr::I<DataVector, VolumeDim, Frame::ElementLogical> xi =
+      interface_logical_coordinates(face_mesh, direction);
+  // logical coordinates on the face mesh in the host block logical coordinates
```

Comment:
You aren't actually mapping element to block logical coords. To understand what's going on here, could you put some explanation / context in these comments? I don't get much information from a comment that just puts the next line of code into English. Something like "Generate logical coords on the block face and its neighbor. To account for the orientation of the neighbor we need to ... because ..."